src/study_groups.py

```
@@ -0,0 +1,712 @@
+
+    try:
+        rows = await env.DB.prepare(sql).bind(*bind_args).all()
+    except Exception as exc:
+        # Avoid logging user-provided content; keep operation and exception only.
+        print(f"[study_groups._list_groups.ERROR] {type(exc).__name__}: {exc}")
+        return helpers["err"]("Failed to list study groups", 500)
+
+    enc = env.ENCRYPTION_KEY
+    groups = []
+    for r in rows.results or []:
+         # description is stored encrypted at rest; decrypt for API response.
```

Flake8 (7.3.0)

[error] indentation is not a multiple of 4 (comment)

(E114)